rust/otap-dataflow/crates/otap/src/otlp_exporter.rs

```diff
@@ -105,25 +106,17 @@ impl Exporter<OtapPdata> for OTLPExporter {
             .start_periodic_telemetry(Duration::from_secs(1))
             .await?;
 
+        let channel = Channel::from_shared(self.config.grpc_endpoint.clone())
+            .map_err(|e| Error::ExporterError {
+                exporter: exporter_id.clone(),
+                error: format!("grpc channel error {e}"),
+            })?
+            .connect_lazy();
```

Comment:
This definitely looks better but I was wondering if this would solve the problem completely. The current export code would still break out of the loop and return from method when it encounters an error.

```rust
 _ = logs_client.export(bytes).await.map_err(|e| {
    self.pdata_metrics.logs_failed.inc();
    Error::ExporterError {
        exporter: exporter_id.clone(),
        error: e.to_string(),
    }
})?;
```

So even if we don't return from the method initially if the server is not available, we might still return from the method later on if there is an error during the actual usage of the channel.

Reply:
Thanks @utpilla! Originally I was planning on handling the failed requests in a followup, but after thinking about it more, I agree it makes sense to fix this all in one PR. I added some changes to gracefully handle the failed requests. It just updating the failed metric for now. Eventually we'll probably have some better behaviour like nacking the message and/or logging the error.